spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTransactionManager.java

```diff
@@ -46,26 +38,23 @@
  * @since 1.1
  */
 public class DatastoreTransactionManager extends AbstractPlatformTransactionManager {
+	private final static TransactionOptions READ_ONLY_OPTIONS = TransactionOptions.newBuilder()
+			.setReadOnly(TransactionOptions.ReadOnly.newBuilder().build())
+			.build();
+
 	private final Supplier<Datastore> datastore;
 
 	public DatastoreTransactionManager(final Supplier<Datastore> datastore) {
 		this.datastore = datastore;
 	}
 
-	Tx getCurrentTX() {
-		return TransactionSynchronizationManager.isActualTransactionActive()
-				? (Tx) ((DefaultTransactionStatus) TransactionAspectSupport
-						.currentTransactionStatus()).getTransaction()
-				: null;
-	}
-
 	@Override
 	protected Object doGetTransaction() throws TransactionException {
-		Tx tx = getCurrentTX();
+		Tx tx = (Tx) TransactionSynchronizationManager.getResource(datastore.get());
```

Comment:
Typically, `getResource` accepts the resource factory instead of the resource itself. Let me explain this by using JDBC terminology: You'd call `getResource(…)` with a `DataSource` and not a `Connection`.

Reply:
In our case, I think `Datastore` is the factory. It might be confusing because we use `Supplier<Datastore>`, but that's just to enable connections to other projects, think different Datastore instances. Most of the time that supplier will always return the same instance. Notice how we call `datastore.newTransaction()`, which essentially creates a new session. Does it make sense?

Reply:
Yeah, the `Supplier<…>` caused some confusion. I'd still recommend adding eventually a dedicated interface (e.g. `DatastoreFactory`, `DatastoreSupplier`) instead of using a raw `Supplier`.

Reply:
Fair point. How about the rest of the `TransactionSynchronizationManager` usage. Does it look good to you?

Reply:
Yes, the rest looks good to me.